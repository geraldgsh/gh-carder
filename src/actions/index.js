const SIGNEDIN = username => ({
  type: 'SIGNEDIN',
  username,
});
const SIGNEDOUT = () => ({ type: 'SIGNEDOUT' });

export { SIGNEDIN, SIGNEDOUT };
