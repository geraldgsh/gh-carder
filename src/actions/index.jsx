const CREATE_USER = 'CREATE_USER';
const REMOVE_USER = 'REMOVE_USER';

const createUser = user => ({
  type: CREATE_USER,
  user,
});

const removeUser = user => ({
  type: REMOVE_USER,
  user,
});

export {
  CREATE_USER, REMOVE_USER, createUser, removeUser,
};
