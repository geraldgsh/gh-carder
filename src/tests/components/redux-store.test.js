import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import '@testing-library/jest-dom/extend-expect';
import configureStore from 'redux-mock-store';
import { createUser } from '../../actions/index';

Enzyme.configure({ adapter: new Adapter() });

const middlewares = [];
const mockStore = configureStore(middlewares);
const store = mockStore({});

describe('Redux store', () => {
  it('should dispatch createUser action', () => {
    store.dispatch(createUser({
      login: 'geraldgsh',
      id: 29500109,
    }));
    const actions = store.getActions();
    expect(actions).toMatchSnapshot();
  });
});
