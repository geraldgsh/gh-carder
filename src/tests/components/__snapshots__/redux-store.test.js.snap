// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redux store should dispatch createUser action 1`] = `
Array [
  Object {
    "type": "CREATE_USER",
    "user": Object {
      "id": 29500109,
      "login": "geraldgsh",
    },
  },
]
`;
